### ISSUE TITLE
App name length limited to 24 characters

### ISSUE THREAD

**Issue by arkkanoid** [deis/workflow#212]
Hi, 
In Deis v1 I used to deploy apps with length > 24 characters, particularly I could deploy and app with a maximum length of 63 chars (The max allowed in a subdomain).
Now in Deis v2 when I try to deploy an app with length> 24 chars I get:

```
    Creating Application... Error: 
    400 Bad Request
    id: Ensure this field has no more than 24 characters.
```

It's possible to increase this length?
Why Deis is limiting to 24 chars?

**Comment by bacongobbler** [comment]
This is due to kubernetes' limitation with service names. https://github.com/kubernetes/kubernetes/issues/12463

**Comment by bacongobbler** [comment]
Note that I will follow up with upstream and see if we can at least get it up to 63 characters again.

**Comment by bacongobbler** [comment]
https://github.com/kubernetes/kubernetes/pull/25041

**Comment by arkkanoid** [comment]
Thanks @bacongobbler !

**Comment by bacongobbler** [comment]
FYI https://github.com/kubernetes/kubernetes/pull/29523 was just merged today so we could potentially add a app name length limit clause for clusters using Kubernetes with that patch. Re-opening so we can track this effort.

**Comment by bacongobbler** [comment]
v1.4.0 now has this new feature. Once we deprecate support for k8s v1.3 we can migrate over to larger app names (being a limit of 63 characters).

**Comment by bacongobbler** [comment]
link: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md/#v140-alpha3

**Comment by tmc** [comment]
for anyone working on this, steps to implement this:

1. migrate the app id SlugField to have higher maximum: https://github.com/deis/controller/blob/master/rootfs/api/models/app.py#L89-L91
2. move length checking into a validator
3. add test showing failure with long lengths
4. expose maximum length setting via a configuration param ala REGISTRATION_MODE

**Comment by bacongobbler** [comment]
a work-in-progress PR is available at #1198 if anyone feels like taking it up and fixing it. Seems like it is mostly working; it's just the unit tests that are failing with the mock scheduler.
